src/test/java/org/lsc/Ldap2LdapOrderedSyncTest.java

```diff
@@ -0,0 +1,190 @@
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *                  ==LICENSE NOTICE==
+ *
+ *               (c) 2008 - 2011 LSC Project
+ *         Sebastien Bahloul &lt;seb@lsc-project.org&gt;
+ *         Thomas Chemineau &lt;thomas@lsc-project.org&gt;
+ *         Jonathan Clarke &lt;jon@lsc-project.org&gt;
+ *         Remy-Christophe Schermesser &lt;rcs@lsc-project.org&gt;
+ ****************************************************************************
+ */
+package org.lsc;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import javax.naming.NamingException;
+import javax.naming.directory.Attribute;
+import javax.naming.directory.SearchResult;
+
+import org.apache.commons.codec.binary.Base64;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.lsc.beans.IBean;
+import org.lsc.configuration.LscConfiguration;
+import org.lsc.exception.LscServiceException;
+import org.lsc.jndi.SimpleJndiSrcService;
+import org.lsc.service.IService;
+import org.lsc.utils.directory.LDAP;
+
+/**
+ * This test case attempts to reproduce a typical ldap2ldap setup via
+ * SimpleSynchronize. This relies on settings in lsc.properties from the test
+ * resources, and entries in the local OpenDS directory. testSync() performs a
+ * synchronization between two branches of the local LDAP server, and should
+ * perform 3 operations : 1 ADD, 1 MODRDN, 1 MODIFY. The
+ * 
+ * @author Jonathan Clarke &ltjonathan@phillipoux.net&gt;
+ */
+public class Ldap2LdapOrderedSyncTest extends CommonLdapSyncTest {
+
+
+	@Before
+	public void setup() {
+		LscConfiguration.reset();
+		LscConfiguration.getInstance();
+		Assert.assertNotNull(LscConfiguration.getConnection("src-ldap"));
+		Assert.assertNotNull(LscConfiguration.getConnection("dst-ldap"));
+		reloadJndiConnections();
+	}
+
+	@Test
+	public final void testLdap2LdapOrderedSyncTest() throws Exception {
+
+		// make sure the contents of the directory are as we expect to begin with
+		List<String> user_tasks = Arrays.asList("ldap2ldapOrderedTestTask3", "ldap2ldapOrderedTestTask2", "ldap2ldapOrderedTestTask1");
+
+		// perform the sync
+		launchSyncCleanTask(user_tasks, false, true, false);
+
+		// check the results of the synchronization
+		reloadJndiConnections();
+		checkSyncResults();
+	}
+
+	private final void checkSyncResults() throws Exception {
+		List<String> attributeValues = null;
+
+
+		// check Operation order
```

Comment:
it seems your test is not checking anything ^^

Reply:
The test itself just launch the tasks in a specific order (3,2,1), while these tasks are defined in an other order in lsc.xml (1,2,3).
But when the tasks are executed in the order (1,2,3), it ends inevitably in an error. (see lsc.xml content)
The only way the tasks can succeed is to execute in order (3,2,1)

Reply:
The other idea I had was to capture the lsc output to check the real order execution, but it seemed more complex to me.

Reply:
ohhhh I totally missed this.

I think it's ok if you add two comments:
1. replace the [main comment](https://github.com/lsc-project/lsc/pull/218#discussion_r1254837017) by an explanation
2. replace `// check Operation order` by a little explanation telling why it fails

Reply:
I have fixed your 2 last points. I will merge the PR